src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenHashTable.cs

```diff
@@ -292,6 +275,28 @@ private static int CalcNumBuckets(ReadOnlySpan<int> hashCodes, bool optimizeForR
             ArrayPool<int>.Shared.Return(seenBuckets);
 
             return bestNumBuckets;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            bool IsBucketFirstVisit(int code)
```

Comment:
Even though this is aggressively inlined, because it's closing over state, the C# compiler is still going to generate a closure struct to store the closed over state.  Does it get optimized by the JIT as much as we'd want?

Reply:
I benchmarked the code and there was no regressions (I've not analyzed the disassembly).